org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/internal/DebugSession.java

```diff
@@ -60,7 +60,9 @@ public void detach() {
 
     @Override
     public void terminate() {
-        vm.exit(0);
+        if (vm.process().isAlive()) {
```

Comment:
the debug session doesn't need check vm is alive or not. Leave the check to the caller.